pkg/api/converterfromosaapi.go

```diff
@@ -70,6 +72,24 @@ func ConvertV1OpenShiftManagedClusterToOpenShiftManagedCluster(oc *v1.OpenShiftM
 			VnetSubnetID: oc.Properties.MasterPoolProfile.VnetSubnetID,
 			Role:         AgentPoolProfileRoleMaster,
 		}
+
+		// init internal AuthProfile structure for conversion
+		cs.Properties.AuthProfile = &AuthProfile{}
+		// check if AuthProfile is all zero value
+		if !reflect.DeepEqual(oc.Properties.AuthProfile, v1.AuthProfile{}) {
```

Comment:
don't understand why the reflect.DeepEqual is here - why not just range over IdentityProviders immediately?

Reply:
`if len(oc.Properties.AuthProfile.IdentityProviders) > 0 ` will do